### PR TITLE
route.md: Add wildcard example code

### PR DESCRIPTION
**Pull request by Enselic**
To developers completely new to axum, it helps a lot to give example code on to get access to the variable path in a handler.

Rendered:

image

Even if the example is not perfect, my hope is that it will be better than nothing. It certainly would have helped me at least.

But feel free to close this PR if the example is too sloppy.

### CODE REVIEW

File: axum/src/docs/routing/route.md [hunk 1]
```diff
@@ -60,6 +60,16 @@ Wildcard captures can also be extracted using [`Path`](crate::extract::Path).
...
+Example:
+
+```rust
+let app = Router::new().route("/*key", get(handler));
+
+async fn handler(path: axum::extract::Path<String>) -> String {
+    path.0
+}
+```
+
```
Comment (davidpdrsn) [inline axum/src/docs/routing/route.md:72]: Lets combine it with the paragraph above

```suggestion
Wildcard captures can also be extracted using [`Path`](crate::extract::Path):

```rust
use axum::{
    Router,
    routing::get,
    extract::Path,
};

let app = Router::new().route("/*key", get(handler));

async fn handler(Path(path): Path<String>) -> String {
    path
}
```

Note that the leading slash is not included, i.e. for the route `/foo/*rest` and
the path `/foo/bar/baz` the value of `rest` will be `bar/baz`.
```

**Comment by Enselic** [comment]
Thanks for the review, I think I fixed your comment now.

**Comment by Enselic** [comment]
I'd like to keep my PR outbox clean, so let's close this for now.

**Comment by jplatte** [comment]
Hey, sorry for the long delay. Other than failing CI (which might fix itself with the next run, let's see), this looks good now!

**Review by jplatte — APPROVED** [review]

File: axum/src/docs/routing/route.md [hunk 2]
```diff
@@ -56,7 +56,22 @@ Note that `/*key` doesn't match empty segments. Thus:
...
+Wildcard captures can also be extracted using [`Path`](crate::extract::Path):
+
+```rust
+use axum::{
+    Router,
+    routing::get,
+    extract::Path,
+};
+
+let app = Router::new().route("/*key", get(handler));
```
Comment (jplatte) [inline axum/src/docs/routing/route.md:68]: CI failure is actually legit, the compiler can't infer the type of `app` here (specifically the generic state type of the router).

```suggestion
let app: Router = Router::new().route("/*key", get(handler));
```

**Pull request merged**